src/pudl/transform/eia_bulk_elec.py

```diff
@@ -0,0 +1,298 @@
+East North Central,ENC,Wisconsin,WI
+East North Central,ENC,Indiana,IN
+East North Central,ENC,Ohio,OH
+East South Central,ESC,Mississippi,MS
+East South Central,ESC,Alabama,AL
+East South Central,ESC,Tennessee,TN
+East South Central,ESC,Kentucky,KY
+Middle Atlantic,MAT,New York,NY
+Middle Atlantic,MAT,Pennsylvania,PA
+Middle Atlantic,MAT,New Jersey,NJ
+Mountain,MTN,Idaho,ID
+Mountain,MTN,Montana,MT
+Mountain,MTN,Wyoming,WY
+Mountain,MTN,Colorado,CO
+Mountain,MTN,Arizona,AZ
+Mountain,MTN,New Mexico,SN
+Mountain,MTN,Utah,UT
+Mountain,MTN,Nevada,NV
+New England,NEW,Maine,ME
+New England,NEW,Rhode Island,RI
+New England,NEW,New Hampshire,NH
+New England,NEW,Vermont,VT
+New England,NEW,Connecticut,CT
+New England,NEW,Massaschusetts,MA
+Pacific Contiguous,PCC,Washington,WA
+Pacific Contiguous,PCC,California,CA
+Pacific Contiguous,PCC,Oregon,OR
+Pacific Noncontiguous,PCN,Alaska,AK
+Pacific Noncontiguous,PCN,Hawaii,HI
+South Atlantic,SAT,West Virginia,WV
+South Atlantic,SAT,Virginia,VA
+South Atlantic,SAT,South Carolina,SC
+South Atlantic,SAT,District of Columbia,DC
+South Atlantic,SAT,Georgia,GA
+South Atlantic,SAT,Florida,FL
+South Atlantic,SAT,Maryland,MD
+South Atlantic,SAT,North Carolina,NC
+South Atlantic,SAT,Delaware,DE
+West North Central,WNC,South Dakota,SD
+West North Central,WNC,North Dakota,ND
+West North Central,WNC,Minnesota,MN
+West North Central,WNC,Nebraska,NE
+West North Central,WNC,Missouri,MO
+West North Central,WNC,Kansas,KS
+West North Central,WNC,Iowa,IA
+West South Central,WSC,Texas,TX
+West South Central,WSC,Louisiana,LA
+West South Central,WSC,Oklahoma,OK
+West South Central,WSC,Arkansas,AR
+    """
+    ),
+)
+"""Association table describing the many-to-one relationships between states and Census Regions used in EIA's aggregates."""
+
+
+def _get_empty_col_names(metadata: pd.DataFrame) -> set[str]:
+    all_nan = metadata.isna().all(axis=0)
+    all_none = metadata.eq("None").all(axis=0)
+    to_drop = all_nan | all_none
+    dropped_col_names = set(metadata.columns[to_drop])
+    expected_to_drop = {
+        "category_id",
+        "childseries",
+        "copyright",
+        "lat",
+        "latlon",
+        "lon",
+        "notes",
+        "parent_category_id",
+    }
+    diff = dropped_col_names.symmetric_difference(expected_to_drop)
+    assert diff == set(), f"Unexpected dropped columns: {diff}"
+    return dropped_col_names
+
+
+def _get_redundant_frequency_col_names(metadata: pd.DataFrame) -> set[str]:
+    freq_map = {
+        "monthly": "M",
+        "quarterly": "Q",
+        "annual": "A",
+    }
+    assert (
+        metadata["frequency_code"].eq(metadata["frequency"].map(freq_map)).all(axis=0)
+    ), "Conflicting information between 'frequency_code' and 'frequency'."
+    assert (
+        metadata["frequency_code"].eq(metadata["f"]).all(axis=0)
+    ), "Conflicting information between 'frequency_code' and 'f'."
+    # keep frequency_code for reference
+    return {
+        "f",
```

Comment:
Yes, it's always the same. Each series is defined by 5 dimensions (metric, fuel, region, sector, frequency) and for some reason the EIA only give frequency its own column (called "f"). The rest are all encoded in string fields (`series_id` and `name`). This function checks that the frequency I parse out from those string fields matches the EIAs "f" column, then removes that column because it is no longer needed. The parsed columns have more legible names ("frequency" and "frequency_code")